carbon-design-system/carbon
# fix(ComboBox): resolves DAP violations

Closes #3260

Resolves DAP violations for ComboBox. 

Current Violations: 
![Screen Shot 2019-09-18 at 11 08 58 AM](https://user-images.githubusercontent.com/17281178/65165688-e6e4f600-da04-11e9-8774-e035356accdb.png)

#### Changelog

**New**

- `tabIndex`, `aria-owns`, `aria-labelledby` on the `input`
- `aria-label` on ListBox 
-  `aria-label` on the Close button svg

**Changed**

- Moved the `aria-label` to the outer container to replicate what accessible combobox examples do, see guidance [here](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html)

#### Testing / Reviewing

- Run DAP on ComboBox, preferably with the dropdown open and something selected because that captures the most violations. 
- Make sure that the only  violations you see are these ones: 

![Screen Shot 2019-09-18 at 11 09 38 AM](https://user-images.githubusercontent.com/17281178/65165796-14ca3a80-da05-11e9-8caa-e736a833f9e4.png)
 those are apparently always there and are a part of storybook 
- check the fixes and make sure they're okay/what should be done. 

*Note*: if you check DAP with the ComboBox closed there will be this violation: 

![Screen Shot 2019-09-18 at 11 14 07 AM](https://user-images.githubusercontent.com/17281178/65165989-77233b00-da05-11e9-8acc-40489d070f62.png
) but this will go away with the ComboBox open as the dropdown list part is what has the id `carbon-combobox-example__menu` and it doesn't exist when the ComboBox is closed
